src/Utility/HDF5Restart.C

```diff
@@ -69,8 +69,13 @@ bool HDF5Restart::writeData (const SerializeData& data)
 #endif
 
   // Lambda function for writing data to the HDF5 file.
-  auto&& writeGroup = [this,pid,ptot] (hid_t group, const std::string& name,
-                                       int len, const void* data, hid_t type)
+#ifdef HAVE_MPI
+  auto&& writeGroup = [this,pid,ptot]
```

Comment:
Hmm, to me it looks like `this` and `ptot` also are referred outside the `#ifdef HAVE_MPI`. Have you tested this with hdf5 support?

Reply:
it's not. it is used in the for immediately after the lambda definition so maybe you got confused by that?

Reply:
right, the use was only outside the lambda I can se now